webcrypto/subtle_crypto.go

```diff
@@ -520,29 +521,38 @@ func (sc *SubtleCrypto) ImportKey(
 		return promise
 	}
 
+	ki, err := newKeyImporter(rt, normalized, algorithm)
+	if err != nil {
+		reject(err)
+		return promise
+	}
+
 	// 5.
 	go func() {
-		switch normalized.Name {
-		case AESCbc, AESCtr, AESGcm:
-			// 8.
-			result, err := importAESKey(format, normalized, keyBytes, keyUsages)
-			if err != nil {
-				reject(err)
-				return
-			}
+		// 8.
+		result, err := ki.ImportKey(format, keyBytes, keyUsages)
+		if err != nil {
+			reject(err)
+			return
+		}
 
-			// 9.
-			if (result.Type == SecretCryptoKeyType || result.Type == PrivateCryptoKeyType) && len(keyUsages) == 0 {
-				reject(NewError(0, SyntaxError, "the keyUsages argument must contain at least one valid usage for the algorithm"))
-				return
-			}
+		// 9.
+		isSecretKey := result.Type == SecretCryptoKeyType
+		isPrivateKey := result.Type == PrivateCryptoKeyType
+		isUsagesEmpty := len(keyUsages) == 0
+		if (isSecretKey || isPrivateKey) && isUsagesEmpty {
+			reject(NewError(0, SyntaxError, "usages cannot not be empty for a secret or private CryptoKey"))
```

Comment:
Is syntax the proper error here?

Reply:
It is indeed, as detailed in step `9.` of the `[importKey](https://www.w3.org/TR/WebCryptoAPI/#SubtleCrypto-method-importKey)` algorithm 👍🏻